mlir/lib/Mitigation/Transforms/MitigationMethods/Zne.cpp

```diff
@@ -243,7 +187,168 @@ FlatSymbolRefAttr ZneLowering::getOrInsertFoldedCircuit(Location loc, PatternRew
     // Remove device
     rewriter.create<quantum::DeviceReleaseOp>(loc);
     rewriter.create<func::ReturnOp>(loc, funcFolded);
-    return SymbolRefAttr::get(ctx, fnFoldedName);
+    return SymbolRefAttr::get(rewriter.getContext(), fnFoldedName);
+}
+// In *.cpp module only, to keep extraneous headers out of *.hpp
+FlatSymbolRefAttr randomLocalFolding(PatternRewriter &rewriter, std::string fnFoldedName,
```

Comment:
Do you want to include the random local folding in this PR?

Reply:
I suggest we leave random local folding out of this PR

Reply:
We'll leave it out, but we'll discuss it for the future, at the next opportunity to bring the teams together.